app/routes/onboarding/userDetails.tsx

```diff
@@ -0,0 +1,235 @@
+import React, { FC, useEffect, useMemo, useState } from 'react';
+import { Form, useSubmit, useActionData } from '@remix-run/react';
+import { ActionFunction, json, redirect, LoaderFunction } from '@remix-run/node';
+import debounce from 'lodash.debounce';
+import axios from 'axios';
+import UserInput from '../../components/common/userInput';
+import Dropdown from '../../components/common/dropdown';
+import { Button } from '../../components/Button';
+
+interface UserFormInterface {
+  username: string;
+  firstname: string;
+  lastname: string;
+  timezone: string;
+}
+
+interface UserDetailsInterface {
+  apiHost: string;
+
+  page: number;
+  formTitlesAndSubtitles: Array<object>;
+}
+
+const isUsernameAvailable = async (host: string, username: string) => {
+  const url = `${host}/api/v1/users/usernameCheck/${username}`;
+  try {
+    const response = await axios.get(url, {
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      withCredentials: true,
+    });
+    return response.data.data.available;
+  } catch (error) {
+    console.error('err', error);
+    return error;
+  }
+};
+
+export const action: ActionFunction = async ({ request, params }) => {
+  const formData = Object.fromEntries(await request.formData());
+
+  const url = `${process.env.API_HOST}/api/v1/users/self`;
+  const { username, firstname, lastname, timezone } = formData;
+
+  const errors = {
+    username: username ? null : 'Username is required',
+    firstname: firstname ? null : 'firstname is required',
+    lastname: lastname ? null : 'lastname is required',
+    timezone: timezone ? null : 'Timezone is required',
+  };
+
+  const hasErrors = Object.values(errors).some((errorMessage) => errorMessage);
+  if (hasErrors) {
+    return json(errors);
+  }
+
+  try {
+    const cookieHeader = request.headers.get('Cookie');
+    await axios.patch(url, formData, {
+      headers: {
+        Cookie: cookieHeader,
+        'Content-Type': 'application/json',
+      },
+    });
+
+    return redirect('/onboarding/userCalendarDetails');
+  } catch (error) {
+    return error;
+  }
+};
+
+const UserDetails: FC<UserDetailsInterface> = () => {
+  const errors = useActionData();
+
+  const initialUserDetails = {
+    username: '',
+    firstname: '',
+    lastname: '',
+    timezone: '',
+  };
+  const UserDetailErrors = {
+    username: '',
+    firstname: '',
+    lastname: '',
+    timezone: '',
+  };
+
+  const [userForm, setUserForm] = useState<UserFormInterface>(initialUserDetails);
+  const [usernameAvailable, setUsernameAvailable] = useState<boolean | undefined>();
+  const [query, setQuery] = useState<string>('');
+  const [userErrors, setUserErrors] = useState(UserDetailErrors);
+
+  const submit = useSubmit();
+
+  useEffect(() => {
+    (async () => {
+      if (query !== '') {
+        const res: boolean = await isUsernameAvailable(window.ENV.API_HOST, query);
+        setUsernameAvailable(res);
```

Comment:
this should be in a `try catch` block as we also return `error`

Reply:
we already have one inside the function

Reply:
in `isUsernameAvailable` function you are returning `error` from catch block. In case of error, how will you catch it here?

Reply:
we can pick this up in the next `PR` as discussed, i will add a comment for this